rebar.config

```diff
@@ -8,5 +8,5 @@
 {project_plugins, [rebar_gleam]}.
 
 {deps, [
-    {gleam_stdlib, "0.11.0"}
+    {gleam_stdlib, "< 0.15"}
```

Comment:
What's the benefit of this? Isn't the particular version in `rebar.lock` the version that gets used, always? If so, why specify this more vaguely?

Reply:
Setting the exact version won't allow using this library in a project with any other `gleam_stdlib` version. Another option would be to define the minimal version `~> 0.11.0` and don't limit it from above, but I prefer to lock on the existing and tested versions.

Reply:
Changed to `~> 0.11.0` after testing the forked version in my project.

Reply:
> Setting the exact version won't allow using this library in a project with any other gleam_stdlib version.

I don't think that's quite right. I just updated another Gleam (Rebar3) project of mine to use `gleam_stdlib` v0.14.0, and it also has a dependency on this project at v1.5.1, which uses of course the older `gleam_stdlib` v0.11.0, and had no issues with it. Check out its [`rebar.config`](https://github.com/rjdellecese/gleam_jsone/blob/v0.4.0/rebar.config) and [`rebar.lock`](https://github.com/rjdellecese/gleam_jsone/blob/v0.4.0/rebar.lock) files to see what I mean.

The section on dependency resolution in the Rebar3 docs says:

> Rebar3 will also use the lock file as the true source of authority for dependencies when switching branches or fetching transitive deps (it will pick data from the lock file if any is available) rather than the rebar.config file.

Which suggests to me that for the sake of users of this library, it doesn't matter what we put in the `rebar.config` file—all that matters is what it resolves to in `rebar.lock`. Assuming that's all correct, I'd prefer to be more specific in `rebar.config`, too!

There's more relevant stuff in the docs, especially in [this section](https://rebar3.org/docs/configuration/dependencies/#fetching-order).

Reply:
@rjdellecese You're right about rebar3, it ignores versions of transitive dependencies and just uses the one defined in the project's lockfile.

Though it is different with mix. By default it will fail with such error message:

```
Failed to use "gleam_stdlib" (version 0.12.0) because
  deps/gleam_decode/rebar.config requires 0.14.0
  mix.exs specifies 0.12.0
```

Of course, you can use `override: true` to skip that check but it's not the best practice.

If you want to try it yourself, there are two branches in my [fork](https://github.com/ImmaculatePine/gleam_decode): `gleam-0.14` (this PR) and `gleam-0.14-hard-lock` (which defines `gleam_stdlib` version as `0.14.0`).

As for someone's motivation to use mix instead of rebar3, unfortunately rebar3 can't handle some complex dependencies (try to install `aws` Elixir package and rebar3 will fail on `decimal`'s `~> 1.0 or ~> 2.0` version).

Reply:
Of course, that makes great sense—thanks for explaining @ImmaculatePine!

This looks great, then—I'll merge and release a little later today.